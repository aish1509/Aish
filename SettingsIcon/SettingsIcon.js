import IconButton from '@material-ui/core/IconButton';
import PropTypes from 'prop-types';
import Settings from '@material-ui/icons/Settings';
import ArrowDropDown from '@material-ui/icons/ArrowDropDown';
import './SettingsIcon.scss';

import React from 'react';

const SettingsIcon = ({onSettingsClick}) => {
	const handleSettingsClick = e => {
		e.preventDefault();
		onSettingsClick();
	};

	return (
		<div className="settings-icon">
			<IconButton onClick={handleSettingsClick}>
				<Settings style={{color: 'white', fontSize: '20px'}} />
				<ArrowDropDown style={{color: 'white', fontSize: '28px'}} />
			</IconButton>
		</div>
	);
};

SettingsIcon.propTypes = {
	onSettingsClick: PropTypes.func.isRequired
};

export default SettingsIcon;
