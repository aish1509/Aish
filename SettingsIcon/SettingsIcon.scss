.settings-icon {
  background: transparent;
  width: 102px;
  height: 100px;
}
