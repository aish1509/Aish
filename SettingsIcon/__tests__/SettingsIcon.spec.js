import React from 'react';
import {shallow} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';
import SettingsIcon from '../index';

describe('<SettingsIcon />', () => {
	let wrapper = null;
	let mockOnSettingsClick = null;
	let mockEvent = null;

	beforeEach(() => {
		mockOnSettingsClick = jest.fn();
		mockEvent = {
			preventDefault: jest.fn()
		};

		wrapper = shallow(
			<SettingsIcon
				lastName="Doe"
				firstName="John"
				onSettingsClick={mockOnSettingsClick}
			/>
		);
	});

	it('should render correctly', () => {
		expect(shallowToJson(wrapper)).toMatchSnapshot();
	});

	it('handleClick', () => {
		wrapper.find('WithStyles(ForwardRef(IconButton))').prop('onClick')(
			mockEvent
		);
		expect(mockEvent.preventDefault).toHaveBeenCalled();
		expect(mockOnSettingsClick).toHaveBeenCalled();
	});
});
