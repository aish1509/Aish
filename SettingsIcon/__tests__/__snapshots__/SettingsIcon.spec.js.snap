// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SettingsIcon /> should render correctly 1`] = `
<div
  className="settings-icon"
>
  <WithStyles(ForwardRef(IconButton))
    onClick={[Function]}
  >
    <SettingsIcon
      style={
        Object {
          "color": "white",
          "fontSize": "20px",
        }
      }
    />
    <ArrowDropDownIcon
      style={
        Object {
          "color": "white",
          "fontSize": "28px",
        }
      }
    />
  </WithStyles(ForwardRef(IconButton))>
</div>
`;
